### PR TITLE
Fix UI anim curve editor issues

### LINKED ISSUE #11616: UI Editor: Curve Editor does not work
**Describe the bug**
The Curve Editor does not have any functional tools and does not accept any inputs. After dock/undock actions the Curve Editor becomes blank (undocked) or displays values for Keys Editor (docked).

**Steps to reproduce**
1. Go to UI Editor.
2. (Optional) Create a sequence and add a few elements to it for better visibility.
3. Select Curve Editor for Animation Editor.
4. Undock the Curve Editor.

**Expected behavior**
The Curve Editor is functional while docked/undocked.

**Actual behavior**
The Curve Editor does not accept any inputs, has no tools, and displays no visual info.

**Screenshots/Video**
CurveEditor

https://user-images.githubusercontent.com/86952591/187180492-8870d644-f2e4-45fb-a1fa-03800aa0836c.mp4

**Found in Branch**
Development (9abaadb)

**Desktop/Device:**
Device: PC
OS: Windows
Version 10
CPU AMD Ryzen 5 3600
GPU Nvidia GeForce RTX 2060 SUPER
Memory 16GB

**Additional context**
Curve Editor in the Track View works correctly.

### PR DESCRIPTION
**Pull request by michabr**
## What does this PR do?

- Fixes curve editor icons not being found
- The curve editor docking widget does not draw when the layout is saved/restored when the curve editor is undocked. This issue seems to be related to UI Editor not using the fancy docking system, so a longer term fix would be to convert UI Editor to use fancy docking. The current workaround in this PR is to dock the sub-panes back up before saving the layout.

The new ViewMode related code was taken from TrackView.

Fixes #11616

## How was this PR tested?

Manual testing closing and re-opening curve widget, UI editor, main editor.

https://user-images.githubusercontent.com/82236305/194137274-78b76731-584a-415a-9158-ac96a8f1b8b0.mp4

https://user-images.githubusercontent.com/82236305/194137294-20bc9b29-d8b8-4369-97d9-176453c945b2.mp4

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: Gems/LyShine/Code/Editor/Animation/UiAnimViewDialog.cpp [hunk 1]
```diff
@@ -1385,6 +1377,22 @@ void CUiAnimViewDialog::keyPressEvent(QKeyEvent* event)
...
+    m_wndKeyPropertiesDock->hide();
+    m_wndCurveEditorDock->hide();
+}
+
+void CUiAnimViewDialog::showEvent([[maybe_unused]] QShowEvent* event)
+{
+    m_wndKeyPropertiesDock->show();
+    if (m_lastMode == ViewMode::Both)
+    {
+        m_wndCurveEditorDock->show();
```
Comment (srikappa-amzn) [inline Gems/LyShine/Code/Editor/Animation/UiAnimViewDialog.cpp:1392]: shouldn't this be also shown if m_lastMode is ViewMode::CurveEditor ?

**Review by srikappa-amzn — COMMENTED** [review]
The repro steps in the video you uploaded and the github issue look different. The widget goes blank immediately after undocking in the github issue. Is that also something these changes fix now?

File: Gems/LyShine/Code/Editor/Animation/UiAnimViewDialog.cpp (hunk 1 above)
Comment (michabr) [reply Gems/LyShine/Code/Editor/Animation/UiAnimViewDialog.cpp:1392]: When the mode is ViewMode::CurveEditor, it does something unique by hiding the dock window and making the curve editor the central widget, so if we showed the curve editor dock here we'd see an empty window.

**Comment by michabr** [comment]
> The repro steps in the video you uploaded and the github issue look different. The widget goes blank immediately after undocking in the github issue. Is that also something these changes fix now?

It used to immediately go blank when undocking the widget if that widget was undocked the last time the UI Editor was closed. Yes, these changes will fix that.

**Pull request merged**